soc/espressif/esp32/Kconfig

```diff
@@ -69,6 +34,23 @@ config ESP32_APPCPU_DRAM
 	help
 	  Defines APPCPU DRAM area in bytes.
 
+config SOC_ESP32_PROCPU
```

Comment:
Why is this here ?
If you want a short name symbol for backwards compatibility, then that is ok, but I believe we should have `SOC_ESP32S3_PROCPU`  and `SOC_ESP32S3_APPCPU` which corresponds to settings in `soc.yml`:
https://github.com/zephyrproject-rtos/zephyr/blob/2851032b1b46aab33f46fe165de36e98107c6e5b/soc/espressif/soc.yml#L15-L18

else it will be hard later to handle the cpu cores in sysbuild, cause sysbuild will only be able to load the board and SoC trees. (`Kconfig.<board>` and `Kconfig.soc` files)

Reply:
@tejlmand This symbol `SOC_ESP32_PROCPU` and `SOC_ESP32_APPCPU` are means to tell build system whether it is been build for one core of the other. AMP support requires this in current implementation. So this is basically a CONFIG to select which functions to run or not. We could, eventually, convert it to another naming and remove  `SOC` from there. Thoughts?

Reply:
This could be replaced by some CONFIG that defines which cpucluster is being currently used during build. Is that possible?

Reply:
> @tejlmand This symbol `SOC_ESP32_PROCPU` and `SOC_ESP32_APPCPU` are means to tell build system whether it is been built for one core or the other. AMP support requires this in current implementation. So this is basically a CONFIG to select which functions to run or not. We could, eventually, convert it to another naming and remove `SOC` from there. Thoughts?

Thanks for explaining.
I think for now the symbols can be kept, but I still think having a `config SOC_ESP32_ESP32S3_PROCPU` and  `config SOC_ESP32_ESP32S3_APPCPU` in the Kconfig soc-tree in addition is preferred as this aligns the Kconfig.soc with the information available through the soc.yml (and board.yml).

Reply:
especially if transitioning to sysbuild for the multi-core / multi-image build, then I assume having this information available in sysbuild will be beneficial.

Note, this thread is not blocking for a +1.
The most important comments for the review are:
https://github.com/zephyrproject-rtos/zephyr/pull/69194#discussion_r1507557481
https://github.com/zephyrproject-rtos/zephyr/pull/69194#discussion_r1507730859

Reply:
@tejlmand  
1) Why `config SOC_ESP32_ESP32S3..`? I mean, why repeating multiple soc name?  
2) Those definitions are already in soc-tree Kconfig:  
`soc/espressif/esp32/Kconfig`
`soc/espressif/esp32s3/Kconfig`

Reply:
> I mean, why repeating multiple soc name?

My bad, seems I got an `ESP32` too much there. Updated ^^^

> Those definitions are already in soc-tree Kconfig:

Yes, in the part that is purely Zephyr Kconfig related.
One part of the many changes in HWMv2, is to provide an self-contained Kconfig tree for the (arch), board (`Kconfig.<board>`) and SoCs (`Kconfig.soc`).
Meaning that you can source the pure Kconfig tree for the hardware model outside of Zephyr.
At first in sysbuild, but in future also for other purposes, like CI / compliance.

If you try to source legacy hardware model Kconfig tree in sysbuild, then you'll get a huge bunch of undefined symbols.
It's simply not possible to do so. It's all or nothing/

HWMv2 provides a self-contained and re-usable Kconfig tree (and that was tree was I referred to as the Kconfig soc-tree). 

> Those definitions are already in soc-tree Kconfig:

and with slight difference that they are without `S3`, so I'm unsure if those symbols are a bit more esp generic, so they cover any (potential) multi-core esp32 family SoC, or if they are specific for the `esp32s3` SoC.